bin/download_submission.py

```diff
@@ -8,20 +8,37 @@
 def get_args():
     """Set up command-line interface and get arguments without any flags."""
     parser = argparse.ArgumentParser()
-    parser.add_argument("submission_id", type=str, help="The ID of submission")
+    parser.add_argument("--submission-id", "-s", type=str, required=True, help="The ID of submission")
+    parser.add_argument("--file-type", "-f", type=str, required=True, help="The type of file the submission should be")
 
     return parser.parse_args()
 
 if __name__ == "__main__":
     args = get_args()
     submission_id = args.submission_id
+    file_type = args.file_type
 
     syn = synapseclient.Synapse()
     syn.login(silent=True)
 
     submission = syn.getSubmission(submission_id, downloadLocation=".")
     entity_type = submission["entity"].concreteType
-    if entity_type != "org.sagebionetworks.repo.model.FileEntity":
-        open("dummy.txt", 'w').close()
+    file_path = submission["filePath"]
+    
+    # TODO: Eventually we want to abstract this logic into the `make_invalid_file` function
+    # in model-to-data's `run_docker.py`, and move that out somewhere else.
+    invalid_file = f"INVALID_predictions.{file_type}"
 
-    print(entity_type)
+    if entity_type != "org.sagebionetworks.repo.model.FileEntity":
```

Comment:
The 

```
        with open(invalid_file, "w") as d:
            d.write(error_msg)
        print(error_msg)
```

lines are repeated for both blocks of your if/elif. You can just pull those out and use the logic to set `err_msg` to avoid this duplication.

Reply:
[Refactored](https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/pull/42/commits/1c749ef82256e17957c90100e2a60d01a472a8ed)